pkg/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,328 @@
+package imageregistry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/apiservers"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/clusteroperator"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
+	v1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var imageRegistryObjName string = "cluster"
+var imageRegistryNamespace string = "openshift-image-registry"
+var imageRegistryCoName string = "image-registry"
+
+// Builder provides a struct for imageRegistry object from the cluster and a imageRegistry definition.
+type Builder struct {
+	// imageRegistry definition, used to create the imageRegistry object.
+	Definition *imageregistryv1.Config
+	// Created imageRegistry object.
+	Object *imageregistryv1.Config
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// Pull retrieves an existing imageRegistry object from the cluster.
+func Pull(apiClient *clients.Settings) (*Builder, error) {
+	glog.V(100).Infof(
+		"Pulling imageRegistry object name: %s", imageRegistryObjName)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &imageregistryv1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: imageRegistryObjName,
+			},
+		},
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("imageRegistry object %s doesn't exist", imageRegistryObjName)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing imageRegistry from cluster.
+func (builder *Builder) Get() (*imageregistryv1.Config, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Pulling existing imageRegistry with name %s from cluster", builder.Definition.Name)
```

Comment:
Please add namespace to log message:
"Pulling existing imageRegistry with name %s, namespace %s from cluster"

Reply:
there is no nsname, there is no namespace